output/csv/consts.go

```diff
@@ -0,0 +1,40 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2016 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// Package csv custom enum types used in config
+package csv
+
+// TimeFormat custom enum type
+type TimeFormat string
+
+// valid defined values for TimeFormat
+const (
+	Unix    TimeFormat = "unix"
+	RFC3399 TimeFormat = "rfc3399"
+)
+
+// IsValid validates TimeFormat
+func (timeFormat TimeFormat) IsValid() bool {
+	switch timeFormat {
+	case Unix, RFC3399:
+		return true
+	}
+	return false
+}
```

Comment:
This seemed to be idiomatic for Go, and allows for a clean extension of `TimeFormat` types in the future.

Reply:
If a timeFormat exists it must be valid, for that I prefer to maintain this logic on the parsing side so we can keep it centralized. @mstoykov opinion?

I think we should move this check in the output constructor so we can check the consolidated value so we can validate also values from json and env.

Reply:
If you can make a suggestion that might help illustrate what you mean?

Reply:
https://github.com/grafana/k6/pull/2274#discussion_r885497705

Reply:
For enums in general we use [enumer](https://github.com/alvaroloes/enumer) as for example with 
https://github.com/grafana/k6/blob/1679c6843c7a5578b25aaba228d98476926467c2/lib/runtime_options.go#L32-L33 

Which generates https://github.com/grafana/k6/blob/1679c6843c7a5578b25aaba228d98476926467c2/lib/compatibility_mode_gen.go and is then used 
https://github.com/grafana/k6/blob/1679c6843c7a5578b25aaba228d98476926467c2/lib/runtime_options.go#L71-L78

Reply:
> For enums in general we use [enumer](https://github.com/alvaroloes/enumer) as for example with
> 
> https://github.com/grafana/k6/blob/1679c6843c7a5578b25aaba228d98476926467c2/lib/runtime_options.go#L32-L33
> 
> Which generates https://github.com/grafana/k6/blob/1679c6843c7a5578b25aaba228d98476926467c2/lib/compatibility_mode_gen.go and is then used
> 
> https://github.com/grafana/k6/blob/1679c6843c7a5578b25aaba228d98476926467c2/lib/runtime_options.go#L71-L78

@mstoykov I spent a long time learning `enumer` and how it works for Go.  I ended up with the follow error:
`enumer: can't handle non-integer constant type TimeFormat`

From what I can understand, this would only work on integer-based enums, whereas in this case we need to preserve the string literals as they are passed in and compared from the user (via the command line).

Let me know if this is still possible with your suggestion, otherwise i'm at a loss how to implement what you are suggesting.